### PR TITLE
refactor: snake case database field for read receipts

### PR DESCRIPTION
**Pull request by atomrc**

### CODE REVIEW

File: app/script/event/preprocessor/ReadReceiptMiddleware.js [hunk 1]
```diff
@@ -20,36 +20,43 @@
...
+   * @returns {Promise<Object>} event - the original event
    */
   processEvent(event) {
     switch (event.type) {
       case z.event.Client.CONVERSATION.CONFIRMATION:
-        this.logger.info(`Received confirmation from '${event.from}' for message '${event.data.message_id}'`);
+        this.logger.info(
+          `Received confirmation of type '${event.data.status}' from '${event.from}' for message '${
+            event.data.message_id
+          }'`
```
Comment (ffflorian) [inline app/script/event/preprocessor/ReadReceiptMiddleware.js:45]: Let's put the message in a variable before `this.logger.info()`.

File: app/script/event/preprocessor/ReadReceiptMiddleware.js [hunk 2]
```diff
@@ -20,36 +20,43 @@
...
+        );
         return this.eventService
           .loadEvent(event.conversation, event.data.message_id)
           .then(originalEvent => {
             if (!originalEvent) {
               return;
             }
+            const currentReadReceipts = originalEvent.read_receipts || [];
+            if (currentReadReceipts.includes(event.from)) {
```
Comment (AndyLnd) [inline app/script/event/preprocessor/ReadReceiptMiddleware.js]:
```suggestion
            if (currentReadReceipts.some(({from}) => event.from === from)) {
```

File: app/script/event/preprocessor/ReadReceiptMiddleware.js [hunk 3]
```diff
@@ -20,36 +20,43 @@
...
-   * If the event does contain a quote, will also decorate the event with some metadata regarding the quoted message
+   * Handles incoming confirmation events
    *
    * @param {Object} event - event in the DB format
-   * @returns {Promise<Object>} event - the original event if no quote is found (or does not validate). The decorated event if the quote is valid
+   * @returns {Promise<Object>} event - the original event
    */
   processEvent(event) {
     switch (event.type) {
       case z.event.Client.CONVERSATION.CONFIRMATION:
```
Comment (ffflorian) [inline app/script/event/preprocessor/ReadReceiptMiddleware.js]: Let's wrap it in `{}`

File: app/script/event/preprocessor/ReadReceiptMiddleware.js [hunk 4]
```diff
@@ -20,44 +20,53 @@
...
+        );
         return this.eventService
           .loadEvent(event.conversation, event.data.message_id)
           .then(originalEvent => {
             if (!originalEvent) {
               return;
             }
+            const currentReadReceipts = originalEvent.read_receipts || [];
+            if (currentReadReceipts.some(({from}) => event.from === from)) {
+              // if the user is already in the readers of the message, nothing more to do
```
Comment (ffflorian) [inline app/script/event/preprocessor/ReadReceiptMiddleware.js]:
```suggestion
              // if the user is already among the readers of the message, nothing more to do
```

**Pull request merged**